### ISSUE TITLE
What parameters will affect the effect of SfM 

### ISSUE THREAD

**Issue by HsiangYangChu** [colmap/colmap#1336]
I found that the pose predicted by `colmap` is not so good  using the `automatic_reconstructor`, so I wanna know what parameters will affect the result of SfM?

**Comment by tsattler** [comment]
All parameters of the feature extractor, the parameters and types of camera models, the parameters of the feature matcher, and the parameters of the mapper.

**Comment by HsiangYangChu** [comment]
@tsattler,
Thanks for your reply!
Which are the most important parameters？

**Comment by tsattler** [comment]
That strongly depends on where the issue in the reconstruction process. Hard to say without any details.

**Comment by HsiangYangChu** [comment]
@tsattler,
Thank you for your patience!
I have encountered two issues, however, it is not convenient to show you my dataset which contains many images, so I am trying to explain more details and upload one typical image in my dataset. First, I test on a synthetic dataset containing 100 images rendered by mitsuba, I found nothing has been reconstructed. 
The ground truth:
![rgb_000000](https://user-images.githubusercontent.com/35549898/142725465-fe108444-d664-4dba-9973-116aaec855c8.png)
The fused.ply showed by meshlab:
![image](https://user-images.githubusercontent.com/35549898/142725508-4101a701-486e-41f3-8829-8798e65c547f.png)
The meshe_poisson.ply showed by meshlab when the trim equals 5:
![image](https://user-images.githubusercontent.com/35549898/142725571-fcbb9d86-e480-4e16-b182-74284bae8df4.png)
I am gonna be crazy, wanna know the reason very much!
Second, for another dataset, I can reconstruct the mesh well, but the pose(or the parameters related to the extrinsic matrix) is not correct

**Comment by tsattler** [comment]
Regarding the image from the first dataset: this is an example of an object that cannot be reconstructed by Colmap (or any other SfM software I am aware of). In order for the sparse reconstruction process to succeed, Colmap needs to be able to reliably detect features. This does not work on shiny / reflective object as the appearance of the object changes too much between viewpoints. Similarly, the dense reconstruction stage has to fail as it cannot establish correspondences between pixels.

Regarding the poses:
* Did you take into account that the poses stored in the Colmap reconstruction are not ordered according to the input order of images?
* Colmap uses the local camera coordinate system used in the computer vision literature (x-axis points to the right, y-axis points downwards, z-axis points forwards / camera is looking down the z-axis). I haven't used Mitsuba yet, but are you sure that it used the same camera coordinate system? In the computer graphics literature and in OpenGL, a different coordinate system is used (x-axis points to the right, y-axis points upwards, camera is looking down the -z-axis).

**Comment by HsiangYangChu** [comment]
@tsattler,
I appreciate your valuable insights!
For the first question, I share your view, so your answer confirmes my view! Thanks, again!
For the second:
- I don't quite understand what you mean. I used the code to extract the pose from the `bin` file provided by a famous repo in Github, which I think is reliable.
- Oh, my god! I have ignored this case! Could you provide a conversion method?
Looking forward to your reply! :)

**Comment by tsattler** [comment]
Regarding the first point, I wanted to point out that it always pays off to check whether the pose you use belongs to the correct image by looking at the corresponding image name.

Regarding the second point: digging out code will probably take me as long as you to derive the transformation yourself. It should not be too hard, it is a simple transformation that you apply after the transformation from global coordinates to local camera coordinates via the pose returned by Colmap.

**Comment by HsiangYangChu** [comment]
@tsattler,
Thanks for the reply!
Regarding the first point, I have checked many times, I am sure that it is not the problem.
Regarding the second point: I have tried to multiply the numbers from the 2th & 3th row of the original 4x4 pose matrix reconstructed by `colmap` by -1, namely, multiply the z & y coordinates in local camera coordinates by -1. Is that right? But, it seems that there is something wrong: nothing has been rendered!!!
